swarm/storage/ldbstore.go

```diff
@@ -805,17 +817,17 @@ func newMockEncodeDataFunc(mockStore *mock.NodeStore) func(chunk Chunk) []byte {
 }
 
 // try to find index; if found, update access cnt and return true
-func (s *LDBStore) tryAccessIdx(ikey []byte, po uint8, index *dpaDBIndex) bool {
+func (s *LDBStore) tryAccessIdx(ikey []byte, po uint8, index *dpaDBIndex) (bool, chan struct{}) {
```

Comment:
i'm not a big fan of planting stuff into method return signatures just in order to ignore it in production.
this should be tested through an injected mock that can transmit that a certain state has been reached so that the test can continue executing. i rather see a constructor overloaded with a possibility to inject a mock than a function that returns a value that production code ignores.

Reply:
also - 
`try to find index; if found, update access cnt and return true` - should this be unified with `MarkAccessed` method? aren't these two actions part of the same atomic action?

Reply:
@justelad okay; look into it
try access does more than stated in the comment
=> I refactor the code to make it more obvious.

For now I decided to keep the two methods separate.

See the latest commit.